File: fabric/src/datagen/java/dev/mayaqq/estrogen/datagen/base/platform/PlatformHelper.java
package dev.mayaqq.estrogen.datagen.base.platform;

import net.minecraft.tags.TagKey;
import net.minecraft.world.item.Item;

public interface PlatformHelper {

    ModPlatform getPlatform();

    String getName(String name);

    TagKey<Item> getCommonTag(String name);
}

File: src/fabric/data/kotlin/dev/mayaqq/estrogen/datagen/platform/PlatformHelper.kt
package dev.mayaqq.estrogen.datagen.platform

import com.google.gson.JsonArray
import net.minecraft.tags.TagKey
import net.minecraft.world.item.Item

interface PlatformHelper {
    val platform: Platform
    fun name(name: String): String
    fun commonTag(name: String): TagKey<Item>
}

interface PlatformRecipeHelper : PlatformHelper {
    fun fluidAmount(amount: Long): Long
    fun isModLoaded(modId: String): EstrogenLoadCondition

    data class EstrogenLoadCondition(val name: String, val condition: JsonArray)
}

enum class Platform {
    FABRIC,
    FORGE,
    COMMON
}